app/src/main/java/com/example/habitsmasher/ui/dashboard/HabitItemAdapter.java

```diff
@@ -0,0 +1,84 @@
+package com.example.habitsmasher.ui.dashboard;
+
+import android.content.Context;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import androidx.annotation.NonNull;
+import androidx.constraintlayout.widget.ConstraintLayout;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.example.habitsmasher.Habit;
+import com.example.habitsmasher.R;
+
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Locale;
+
+public class HabitItemAdapter extends RecyclerView.Adapter<HabitItemAdapter.HabitViewHolder> {
+    private static final String DATE_PATTERN = "EEE, MMM d, yyyy";
+    private static final Locale LOCALE = Locale.CANADA;
+
+    private final Context _context;
+    private static ArrayList<Habit> _habits;
+
+    public HabitItemAdapter(Context context, ArrayList<Habit> habits) {
+        _context = context;
+        _habits = habits;
+    }
+
+    @NonNull
+    @Override
+    public HabitViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
+        View view = LayoutInflater.from(_context).inflate(R.layout.habit_row, parent, false);
+        return new HabitViewHolder(view);
+    }
+
+    @Override
+    public void onBindViewHolder(@NonNull HabitViewHolder holder, int position) {
```

Comment:
In the `onBindViewHolder`, all the front-end attributes that make up a habit row are set.